packages/xstate-wallet/src/messaging.ts

```diff
@@ -1,67 +1,117 @@
+import {EventEmitter} from 'eventemitter3';
 import {
-  UpdateChannelParams,
-  CloseChannelParams,
-  CreateChannelParams,
-  PushMessageParams,
-  JoinChannelParams,
   parseRequest,
   CreateChannelRequest,
   UpdateChannelRequest,
-  PushMessageRequest,
-  CloseChannelRequest
+  CloseChannelRequest,
+  JoinChannelRequest,
+  Response,
+  ChannelResult,
+  ChannelStatus,
+  Notification,
+  ChannelClosingNotification,
+  ChannelUpdatedNotification
 } from '@statechannels/client-api-schema';
-import {
-  getChannelId,
-  Channel,
-  CreateChannelEvent,
-  AddressableMessage,
-  unreachable
-} from '@statechannels/wallet-protocols';
-import * as ethers from 'ethers';
-import {bigNumberify} from 'ethers/utils';
+import {unreachable} from '@statechannels/wallet-protocols';
 import * as jrs from 'jsonrpc-lite';
 
-import {
-  createStateVarsFromUpdateChannelParams,
-  createJsonRpcAllocationsFromOutcome
-} from './utils/json-rpc-utils';
-import {WorkflowManager} from './workflow-manager';
 import {fromEvent, Observable} from 'rxjs';
-import {map} from 'rxjs/operators';
-import {filterByPromise} from 'filter-async-rxjs-pipe';
 import {Store} from './store';
 import {ChannelStoreEntry} from './store/memory-channel-storage';
+import {Message} from './store/wire-protocol';
+import {createJsonRpcAllocationsFromOutcome} from './utils/json-rpc-utils';
+
+type ChannelRequest =
+  | CreateChannelRequest
+  | JoinChannelRequest
+  | UpdateChannelRequest
+  | CloseChannelRequest;
+
+interface InternalEvents {
+  ChannelRequest: [ChannelRequest];
+  CreateChannelRequest: [CreateChannelRequest];
+  SendMessage: [Response | Notification];
+}
 
-export function observeRequests(
-  channelId: string
-): Observable<JoinChannelParams | CloseChannelParams | UpdateChannelParams> {
-  return fromEvent(window, 'message').pipe(
-    filterByPromise(async (e: MessageEvent) => {
-      if (!e || !e.data.jsonrpc || e.data.jsonrpc !== '2.0') {
-        return false;
-      }
-      const parsedMessage = jrs.parseObject(e.data);
-      if (parsedMessage.type !== 'request') {
-        return false;
-      }
-      const request = parseRequest(e.data);
-      if (
-        request.method !== 'UpdateChannel' &&
-        request.method !== 'CloseChannel' &&
-        request.method !== 'JoinChannel'
-      ) {
-        return false;
-      }
-      return request.params.channelId === channelId;
-    }),
-    map((e: MessageEvent) => {
-      return e.data.params;
-    })
+export interface MessagingServiceInterface {
```

Comment:
Wondering if it's worth having an interface here? It seems like we'll only have one `MessagingService`, so we could just use that everywhere.

Reply:
Probably, but having a interface makes me feel safer :)